eip_7594/src/lib.rs

```diff
@@ -343,6 +372,64 @@ fn kzg_commitment_inclusion_proof<P: Preset>(
     proof
 }
 
+/**
+ * Return the sample count if allowing failures.
+ *
+ * This helper demonstrates how to calculate the number of columns to query per slot when
+ * allowing given number of failures, assuming uniform random selection without replacement.
+*/
+pub fn get_extended_sample_count(allowed_failures: u64) -> u64 {
+    // check that `allowed_failures` within the accepted range [0 -> NUMBER_OF_COLUMNS // 2]
+    assert!((0..(NumberOfColumns::U64 / 2)).contains(&allowed_failures));
+
+    // missing chunks for more than a half is the worst case
+    let worst_case_missing = NumberOfColumns::U64 / 2 + 1;
+
+    // the probability of successful sampling of an unavailable block
+    let false_positive_threshold = hypergeom_cdf(
+        0,
+        NumberOfColumns::U64,
+        worst_case_missing,
+        SAMPLES_PER_SLOT,
+    );
+
+    // number of unique column IDs
+    let mut sample_count = SAMPLES_PER_SLOT;
+    while sample_count <= NumberOfColumns::U64 + 1 {
+        // TODO(feature/das): change variable name `x` to a suitable one
+        let x = hypergeom_cdf(
+            allowed_failures,
+            NumberOfColumns::U64,
+            worst_case_missing,
+            sample_count,
+        );
+        if x <= false_positive_threshold {
+            break;
+        }
+        sample_count += 1;
+    }
+    return sample_count;
+}
+
+fn hypergeom_cdf(k: u64, m: u64, n: u64, big_n: u64) -> f64 {
+    let mut sum = 0_f64;
+    for i in 0..(k + 1) {
+        sum += math_comb(n, i) * math_comb(m - n, big_n - i) / math_comb(m, big_n);
+    }
+    sum
+}
+
+fn math_comb(n: u64, k: u64) -> f64 {
+    if !(0..n).contains(&k) {
+        return 0_f64;
+    }
+    let mut r = 1_f64;
+    for i in 0..(std::cmp::min(k, n - k)) {
+        r *= (n - i) as f64 / (i + 1) as f64
+    }
+    r
+}
```

Comment:
I believe edge cases should be handled as follows

```rust
fn math_comb(n: u64, k: u64) -> f64 {
    if k > n {
        return 0_f64;
    }
    if k == 0 || k == n {
        return 1_f64;
    }
    let k = std::cmp::min(k, n - k);
    let mut r = 1_f64;
    for i in 0..k {
        r *= (n - i) as f64 / (i + 1) as f64;
    }
    r
}
```

Reply:
> * the range should be 0..=n, since k can be equal to n

nice catch, thank you
> * handle edge cases when k is 0 or n (optional, just for sake of clarity / readability)

found a rust util that equivalent to `math.comb` in Python: https://docs.rs/math_l/latest/src/math_l/math.rs.html#32-38
```rust
// The number of combination: k-combination of n
//  - n: a set size of n
//  - k: taken k of elements from the set
// nCk = n! / ((n - k)! * k!)
pub fn comb(n: i32, k: i32) -> i32 {
    if k > n {
        0
    } else {
        (1..=k).fold(1, |acc, val| acc * (n - val + 1) / val)
    }
}
```